### DISCUSSION TITLE
How to specify `includeDirs` in workflow.yml

### DISCUSSION THREAD

**Discussion by mpdbar** [stepci/stepci#233] [Q&A]
Hi,

first of all, let me thank you for this amazing tool you've built. I am currently evaluating tools for automating functional tests for a gRPC API and so far, stepci is my favorite candidate!

I've followed the tutorial on testing gRPC with stepci and was able to successfully make a gRPC request to a service which does not use any imports. However, calling a service which imports `google/api/annotations.proto` always fails because this dependency cannot be resolved.

Now I know from using other tools like `grpcurl` that I need to specify additional directories (like with option `--import-path` in `grpcurl`); however, there does not appear to be a way to do this in stepci (but there *should* be from what I've gathered from discussions [here](https://github.com/stepci/stepci/discussions/19) and [here](https://github.com/stepci/stepci/discussions/127)).

My knowledge of JS/TS/NodeJS is limited at best, but it seems that `cool-grpc` always passes hard coded `loaderOptions` to `proto-loader`. I managed to do a local workaround by "enabling" YAML fields `config.grpc.includeDirs` and `tests.<test>.steps.[step].grpc.includeDirs` which behave like  `config.grpc.proto` and `tests.<test>.steps.[step].grpc.proto` and "merging" this with the hard coded `defaultLoaderOptions`, but I was wondering if there is another, cleaner way to accomplish this.

Best regards
Moritz

**Comment by mishushakov** [comment]
Would you mind opening a pull request with your changes?

**Reply by mpdbar** [reply]
I absolutely wouldn't mind, but since that would be my first pull request in TypeScript (as in: *ever*), expect some ugly code. Also, that would be 2 PR's since I modified `stepci/runner` and `cool-grpc`.
